Feature sobolev smoothing solver

thomasdick opened the pull request:
This pull request is related to the introduction of a Sobolev smoothing solver. See [su2code/SU2/pull/1446](https://github.com/su2code/SU2/pull/1446/)

It introduces the files necessary to run regression tests for the new solver.

pcarruscag commented on grad_smooth/naca0012/of_hess.dat.ref:
```diff
@@ -0,0 +1,38 @@
+18.98806831, 11.5005011, 6.280203673, 2.929292841, 0.8777531067, -0.3021857211, -0.9109241251, -1.155996628, -1.17939658, -1.077130493, -0.9129959149, -0.7282878372, -0.5486463758, -0.3888924872, -0.2564490893, -0.1537705561, -0.08007565817, -0.03258319132, -0.007379490979, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
```
Put the reference results in the code repo instead please (main PR), files diffs are sensitive and it's easier to update them that way.

pcarruscag reviewed (commented):
Can you use symlinks to some of the files? e.g. meshes

thomasdick commented:
I moved the reference files for filediff in the code repo.
The NACA0012 mesh is now a symlink.
For the ONERA M6 test case I added my own FFD box to the mesh from the compressible flow tutorial. So I can not link to an existing file.
